lib/ueberauth/strategy/linear.ex

```diff
@@ -16,22 +17,19 @@ defmodule Ueberauth.Strategy.Linear do
   Handles initial request for Linear authentication.
   """
   def handle_request!(conn) do
-    scopes = conn.params["scope"] || option(conn, :default_scope)
-    opts = [scope: scopes]
-
-    opts =
-      if conn.params["state"],
-        do: Keyword.put(opts, :state, conn.params["state"]),
-        else: opts
-
     callback_url = callback_url(conn)
 
     callback_url =
       if String.ends_with?(callback_url, "?"),
         do: String.slice(callback_url, 0..-2),
         else: callback_url
 
-    opts = Keyword.put(opts, :redirect_uri, callback_url)
+    opts =
+      []
+      |> with_scopes(conn)
+      |> with_state_param(conn)
```

Comment:
hmm I dont see with_state_param declared, I mightve missed it is it in another file?

Reply:
library function that is imported